plugins/modules/mso_service_node_type.py

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2022, Shreyas Srish (@shrsr) <ssrish@cisco.com>
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: mso_service_node_type
+short_description: Manage Service Node Types
+description:
+- Manage Service Node Types on Cisco ACI Multi-Site.
+author:
+- Shreyas Srish (@shrsr)
+options:
+  name:
+    description:
+    - The name of the node type.
+    type: str
+  display_name:
+    description:
+    - The name of the node type as displayed on the MSO web interface.
+    type: str
+  state:
+    description:
+    - Use C(present) or C(absent) for adding or removing.
+    - Use C(query) for listing an object or multiple objects.
+    type: str
+    choices: [ absent, present, query ]
+    default: present
+extends_documentation_fragment: cisco.mso.modules
+'''
+
+EXAMPLES = r'''
+- name: Add a new Service Node Type
+  cisco.mso.mso_schema_service_node:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    name: Firewall
```

Comment:
I would use something else than Firewall as this is a default type. (same comment for all other examples)